src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt

```diff
@@ -17,6 +17,12 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
 
   fun findByReference(reference: String): Visit?
 
+  @Query(
+    "SELECT v FROM Visit v " +
+      "WHERE v.reference = :reference and v.visitStatus = 'BOOKED'"
+  )
+  fun findBookedVisitByReference(reference: String): Visit?
```

Comment:
Would there be value in passing the status as a parameter or as a List of VisitStatuses and doing an IN in the query?

Reply:
I tend to only put features or logic in when I need it rather than if I need it!

Reply:
I'm not convinced the URL is constructed correctly for this

I would expect it to be either:

/visits/{visit reference}/booked
or
/visits/{visit reference}?status=booked